### PR TITLE
Force highest TLS version supported

### PR DESCRIPTION
**Pull request by sanmai-NL**
The integrity and confidentiality of the installer script currently hinges on TLS. It is important to enforce the highest version of TLS in the instructions. Also, enforce the `https` scheme. Should redirects occur in the future, then each URL redirected to must be accessed using TLS 1.2 with HTTP, rather than allowing a plain HTTP link in the chain.

### CODE REVIEW

**Comment by kinnison** [comment]
Hi @sanmai-NL,

Thank you for your contribution to Rustup.  Would it also make sense to adjust `rustup-init.sh` similarly?

**Comment by sanmai-NL** [comment]
Yes, and https://www.rust-lang.org/learn/get-started as well.

As for backward compatibility of a similar change to `rustup-init.sh`: I cannot determine when `--tlsv1.2` was added to cURL. It could have been added in 2001 already, but I'm not sure based on the [cURL changelog](https://curl.haxx.se/changes.html). I have successfully used these CLI options for about 5 years on production systems, though.

**Comment by kinnison** [comment]
If you can verify that the cURL version in RHEL 7 and later distros is suitable then please update your patch to cover rustup-init.sh and we'll probably merge it.  You'll need a separate issue opening on the website repo for that bit though.

**Comment by sanmai-NL** [comment]
[Apparently RHEL 7 has an up-to-date cURL](https://curl.haxx.se/dlwiz/?type=bin&os=Linux&flav=Redhat&ver=RHEL7).

**Comment by sanmai-NL** [comment]
@kinnison: Done. Also added similar options to the `wget` invocation. According to the changelog, [they support these parameters since 2014](https://fossies.org/linux/wget/ChangeLog).

**Comment by tesuji** [comment]
~I afraid that we cannot use rustup in Ubuntu 14.04 (at least in Travis CI).~
Edit: Looks like I am wrong. [curl manpages of Ubuntu 14.04](https://manpages.ubuntu.com/manpages/trusty/en/man1/curl.1.html) lists those options.

**Comment by sanmai-NL** [comment]
As for cURL, `--tlsv1.2` was introduced in 7.34 and `--proto` in 7.20.2 according to https://curl.haxx.se/docs/manpage.html. Ubuntu 14.04 has [an official cURL package based on 7.35](https://launchpad.net/ubuntu/trusty/+source/curl). You can choose [a more up-to-date Ubuntu edition (Xenial)](https://docs.travis-ci.com/user/reference/xenial/) on Travis CI.

**Comment by kinnison** [comment]
We try to maintain an older CI baseline because it increases our reachability wrt. distributions - glibc versions are a pain.

**Comment by sanmai-NL** [comment]
@kinnison: alright, but this is ready for merging no?

**Review by kinnison — APPROVED** [review]
LGTM, if noone objects in terms of support for other OSes, I'll merge in the next few days.

**Comment by sanmai-NL** [comment]
@kinnison: ping.

**Comment by kinnison** [comment]
@sanmai-NL Our working group meeting is tonight, I planned to check there before hitting merge.

**Comment by tesuji** [comment]
@kinnison I've just noticed that command may not work on zsh. You have to type:
```console
curl --proto \=https --tlsv1.2 -sSf https://sh.rustup.rs
```
or 
```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs
```
instead.

**Comment by kinnison** [comment]
Hmm, I wonder why I didn't spot that..

*rummages*

Looks like I have `setopt no_equals` in my configs.  Well spotted.  Will you provide a PR to fix that please, and I'll merge it.

File: www/index.html
Comment (alexcrichton) [inline www/index.html:39]: Could this be left as it was? This is just intended to be a quick one-liner copy/paste snippet and expanding with so many options can be pretty intimidating as a copy/paste snippet. Does this really add enough benefit relative to what it's displaying on the site?

File: rustup-init.sh
Comment (alexcrichton) [inline rustup-init.sh:379]: It looks like this may be [causing issues on Travis](https://travis-ci.com/rust-lang/cargo/jobs/195079192)?

File: www/index.html
Comment (sanmai-NL) [reply www/index.html:39]: You are expressing certain feelings towards this without backing them up with a UX study. I am sad you prioritize the length of an already technical one-liner over the clear and specified security improvement. I think it is instead a good signal that in the Rust ecosystem, security is taken seriously.

File: rustup-init.sh
Comment (sanmai-NL) [reply rustup-init.sh:379]: We can't tell why without debug logging. We don't know whether it is solely because of this change or due to something else, such as the state of the infra.

Comment (kinnison) [reply rustup-init.sh:379]: This has been resolved as safely as we know how with 1.18.1

File: www/index.html
Comment (kinnison) [reply www/index.html:39]: Certainly the display isn't as good as it used to be.  We *could* use web techniques to hide the extra arguments from the display but include them in the copy/paste, and provide a copy-command-to-clipboard button which puts it all there, for ease of use?

Comment (sanmai-NL) [reply www/index.html:39]: Why would that be important?

File: rustup-init.sh
Comment (sanmai-NL) [reply rustup-init.sh:379]: @kinnison: Can you please explain a bit more?

File: www/index.html
Comment (kinnison) [reply www/index.html:39]: Currently the box requires scrolling, and people are more likely to fail to copy the whole command than not.  While I personally dislike the narrow styling, I understand why it's there so would prefer to maintain it over trying to do better myself.  If someone else could make it work somehow then that'd be good too.

**Pull request merged**
